### ISSUE TITLE
Split Yolov5 in GPU/CPU versions

### ISSUE THREAD

**Issue by jacobowitz** [jina-ai/executors#200]
While working on GPU support for a couple of executors in this [issue](https://github.com/jina-ai/executors/issues/169), I've stumbled over a problem for the [Yolov5 segmenter](https://github.com/jina-ai/executors/blob/main/jinahub/segmenters/YoloV5Segmenter/yolov5_segmenter.py), which can be seem in this [draft pr](https://github.com/pulls).
CI always fails to install pycocotools on our custom [GPU github action runners](https://github.com/jina-ai/jina-terraform/tree/main/gpu-github-runners).
That is fairly annoying, because we dont even need this dependency, it is only introduced because the original upstream [authors of yolov5](https://github.com/ultralytics/yolov5) dont provide a Pypi package and we are using this [fork](https://github.com/fcakyon/yolov5-pip) instead.

So this issue contains two sub tasks:

- [ ] Ideally submit a PR to the original yolov5 repo and provide a setup.py
- [ ] Replace the yolov5 fork with the real one and do necessary changes for separing gpu/cpu version of the yolov5 segmenter

**Comment by hmen97** [comment]
Hi @jacobowitz it looks like this [pull request](https://github.com/ultralytics/yolov5/pull/465) has added all the changes required for a setup.py but the authors are holding off on integrating it as it would involve adding a yolov5 base directory, so they are looking for other options. Meanwhile the author has mentioned [this](https://pypi.org/project/yolov5/) 3rd party Pypi package as the best way to do inference right now.

**Comment by jacobowitz** [comment]
@hmen97 thank you for digging into this issue! Its a pity that yolov5 authors won't refactor their repo to enable a proper Pypi package :(
I think adding their repo as a submodule is not an option as it would break a lot of automation and also does not fit in this general purpose repository here. I think another option might be to fork the fork we are using and change the requirements.txt to a more minimal one. Maybe a better option would be a PR against the existing fork to structure the requirements.txt a bit better? I think the unnecessary dependencies should be marked as such there.

**Comment by hmen97** [comment]
Yeah I was thinking along the same lines, based on what you have said

> CI always fails to install pycocotools on our custom GPU github action runners.

If I make a PR against the forked repo so that unecessary dependencies like pycocotools become optional/can be skipped when running setup.py, I think that should solve the issue, but in case they don't accept that change then I'll fork the fork. Is that alright?

**Comment by jacobowitz** [comment]
@hmen97 that sounds great!

**Comment by CatStark** [comment]
Hi @hmen97 did you have any chance to look into this? it this ticket still relevant to you? :)

**Comment by cristianmtr** [comment]
Hey @hmen97 

We've unassigned you from the issue for now, and moved the issue back to our backlog. If you want to continue your work on it, let me know.
